proxylib/accesslog/client.go

```diff
@@ -28,23 +28,24 @@ import (
 )
 
 type Client struct {
-	connected uint32     // Accessed atomically without locking
-	mutex     lock.Mutex // Used to protect opening the connection
+	connected uint32 // Accessed atomically without locking
 	path      string
+	mutex     lock.Mutex     // Used to protect opening the connection
```

Comment:
If this is only protecting the `unsafe.Pointer` bellow consider using the https://golang.org/pkg/sync/atomic/#LoadPointer and https://golang.org/pkg/sync/atomic/#StorePointer and user a wrapper function like `GetConn` `SetConn`

Reply:
We already use `LoadPointer` and `StorePointer`, but I also wanted to make sure we do not attempt to open two connections to the access log server at the same time. So there is a larger critical section the mutex protects, consisting of
- read pointer value
- connect to access log server unix domain socket
- store pointer value
- store a boolean indicating the that connection was successfully opened

The above critical section needs to be entered only after send failures, otherwise the access logging takes no locks even when the connection is shared between multiple threads.